docs/node/workflows.md

```diff
@@ -57,62 +57,83 @@ It can be used in any NodeJS process e.g an express app and does not depend on t
+  - Equivalent to `new CancellationScope({ cancellable: false }).run(fn)`.
+- [`withTimeout(timeoutMs, fn)`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#withtimeout): If timeout triggers before `fn` resolves the scope will be cancelled, triggering cancellation of enclosed operations, such as activities and timers.
+  - Equivalent to `new CancellationScope({ cancellable: true, timeout: timeoutMs }).run(fn)`.
 
-In the Node SDK, Workflows are represented internally by a tree of scopes. The main function runs in the root scope.
-Cancellation propagates from outer scopes to inner ones and is handled by catching `CancelledError`s
-thrown by cancellable operations (see below).
+Cancellation propagates from outer scopes to inner ones and is handled by catching `CancelledError`s thrown by cancellable operations (see below).
 
-Scopes are created using the [`CancellationScope`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope) constructor or the static helper methods
-[`cancellable`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#cancellable-1),
-[`nonCancellable`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#noncancellable),
-and [`withTimeout`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#withtimeout).
+Scopes all return native JS Promises, so you can use all the familiar Promise APIs like `Promise.all` and `Promise.race` to model your async logic.
+Other APIs you can use:
 
-When a `CancellationScope` is cancelled, it propagates cancellation in any child scopes and of any cancellable operations created within it, such as:
+- `CancellationScope.current()`: get the current scope
+- `scope.cancel()`: cancel all operations inside a `scope`
+- `scope.run()`: run an async function within a `scope`
+- `scope.cancelRequested`: a promise that resolves when a scope cancellation is requested, e.g when Workflow code calls `cancel()` or the entire Workflow is cancelled by an external client.
+
+When a `CancellationScope` is cancelled, it propagates cancellation to any child scopes and any cancellable operations created within it, such as:
 
 - Activities
 - Timers (created with the [`sleep`](https://nodejs.temporal.io/api/modules/workflow#sleep) function)
 - [`Trigger`](https://nodejs.temporal.io/api/classes/workflow.trigger)s
 
-## Examples
-
-### Cancel a timer from Workflow code
+### Internal Cancellation example
 
 <!--SNIPSTART nodejs-cancel-a-timer-from-workflow-->
 <!--SNIPEND-->
 
-### Alternatively, the preceding can be written as
+Alternatively, the preceding can be written as:
 
 <!--SNIPSTART nodejs-cancel-a-timer-from-workflow-alternative-impl-->
 <!--SNIPEND-->
 
-### Run multiple activities with a single deadline
+### External Cancellation example
 
-<!--SNIPSTART nodejs-multiple-activities-single-timeout-workflow-->
+Handle Workflow cancellation by an external client while an Activity is running:
```

Comment:
Perhaps we should add a sample here of how this Workflow could be cancelled using a `WorkflowStub` (doesn't have to be in this PR).

Reply:
```suggestion
Handle Workflow cancellation by an external client while an Activity is running:

<!-- TODO: add a sample here of how this Workflow could be cancelled using a WorkflowStub -->
```